ios/src/CountlyReactNative.m

```diff
@@ -110,6 +103,115 @@ + (BOOL)requiresMainQueueSetup
     });
 }
 
+- (void) populateConfig:(id) json {
+    // NSLog(@"%@",json[@"serverURL"]);
+    if (config == nil) {
+      config = CountlyConfig.new;
+    }
+
+    NSString *serverurl = json[@"serverURL"];
+    NSString *appkey = json[@"appKey"];
+    NSString *deviceID = json[@"deviceID"];
+    if (deviceID != nil && deviceID != (NSString *)[NSNull null] && ![deviceID isEqual:@""]) {
+      config.deviceID = deviceID;
+    }
+    config.appKey = appkey;
+    config.host = serverurl;
+
+    if (json[@"loggingEnabled"]) {
+        config.enableDebug = YES;
+        config.internalLogLevel = CLYInternalLogLevelVerbose;
+    } else {
+        config.enableDebug = NO;
+    }
+
+    if (json[@"shouldRequireConsent"]) {
+        config.requiresConsent = YES;
+    }
+
+    if (json[@"tamperingProtectionSalt"]) {
+        config.secretSalt = json[@"tamperingProtectionSalt"];
+    }
+
+    if (json[@"consents"]) {
+        config.consents = json[@"consents"];
+    }
+
+    if (json[@"starRatingTextMessage"]) {
+        config.starRatingMessage = json[@"starRatingTextMessage"];
+    }
+    // parameter not found: starRatingTextTitle
+    // parameter not found: starRatingTextDismiss
+
+    if (json[@"enableApm"]) {
+        config.enablePerformanceMonitoring = YES;
+    }
+
+    if (json[@"crashReporting"]) {
+        [self addCountlyFeature:CLYCrashReporting];
+    }
+
+#ifndef COUNTLY_EXCLUDE_PUSHNOTIFICATIONS
+    if (json[@"pushNotification"]) {
+        config.sendPushTokenAlways = YES;
+        config.pushTestMode = CLYPushTestModeProduction;
+        NSString *tokenType = json[@"tokenType"];
+        if ([tokenType isEqualToString:@"1"]) {
+            config.pushTestMode = CLYPushTestModeDevelopment;
+        } else if ([tokenType isEqualToString:@"2"]) {
+            config.pushTestMode = CLYPushTestModeTestFlightOrAdHoc;
+        }
+
+        CountlyPushNotifications.sharedInstance.pushTestMode = config.pushTestMode;
+        // parameter not found: channelName
+        // parameter not found: channelDescription
+    }
+#endif
+
+    if (json[@"attributionID"]) {
+          NSString *attributionID = json[@"attributionID"];
+          if (CountlyCommon.sharedInstance.hasStarted) {
+              [Countly.sharedInstance recordAttributionID:attributionID];
+          } else {
+              config.attributionID = attributionID;
+          }
+    }
+    // parameter not found: enableAttribution
+    // parameter not found: allowedIntentClassNames
+    // parameter not found: allowedIntentPackageNames
+
+    if (json[@"locationCountryCode"]) {
+        NSString *countryCode = json[@"locationCountryCode"];
+        NSString *city = json[@"locationCity"];
+        NSString *locationString = json[@"locationGpsCoordinates"];
+        NSString *ipAddress = json[@"locationIpAddress"];
+
+        if (locationString != nil && ![locationString isEqualToString:@"null"] && [locationString containsString:@","]) {
```

Comment:
use 'getCoordinate' function if possible.
https://github.com/Countly/countly-sdk-react-native-bridge/blob/1e98a4ea22c97c40ad54fb624c96a72623deadf2/ios/src/CountlyReactNative.m#L578

Reply:
Please I do not understand

Reply:
You are creating a "CLLocationCoordinate2D" object by parsing the lat log from gps string, we already have a function for this, you can use that function instead of doing it by yourself.